DNN Platform/Library/Services/Analytics/GoogleAnalyticsEngine.cs

```diff
@@ -94,5 +94,52 @@ public override string RenderScript(string scriptTemplate)
             scriptTemplate = scriptTemplate.Replace("[CUSTOM_SCRIPT]", RenderCustomScript(config));
             return scriptTemplate;
         }
+
+        public override string RenderCustomScript(AnalyticsConfiguration config)
+        {
+            try
+            {
+                bool anonymize = false;
+                bool trackingUserId = false;
+
+                foreach (AnalyticsSetting setting in config.Settings)
+                {
+                    switch (setting.SettingName.ToLowerInvariant())
+                    {
+                        case "anonymizeip":
+                        {
+                            bool.TryParse(setting.SettingValue, out anonymize);
+                            break;
+                        }
+                        case "trackinguser":
+                        {
+                            bool.TryParse(setting.SettingValue, out trackingUserId);
+                            break;
+                        }
+                    }
+                }
+
+
+                var sb = new System.Text.StringBuilder();
+                if (anonymize)
+                {
+                    sb.Append("ga('set', 'anonymizeIp', true);");
+                }
+
+                if (trackingUserId)
+                {
+                    sb.AppendFormat("ga('set', 'userId', {0});", UserController.Instance.GetCurrentUserInfo().UserID);
```

Comment:
should this check if they're logged in?  or do we want to track `-1`?

Reply:
I grabbed that code from my old GA extension, and it was running for all users in that extension. 

However, I can see pros/cons for either way. I'm good with -1 for guests or adding a check  IsAuthenticated prior to setting the UserId var in GA. Whichever sounds best to the @dnnsoftware/approvers i'll make happen with another commit.